tests/libcxx/enc/CMakeLists.txt

```diff
@@ -68,6 +68,32 @@ function(add_libcxx_test_enc NAME CXXFILE)
 
     target_compile_definitions(libcxxtest-${NAME}_enc PRIVATE -DWITH_MAIN -D__TEST__="${CXXFILE}")
     target_link_libraries(libcxxtest-${NAME}_enc libcxxtest-support)
+
+    if (WIN32)
+        maybe_build_using_clangw(libcxxtest-${NAME}_enc)
+
+        # maybe_build_using_clangw populates variables in its parent scope (ie current scope)
+        # Propagate these variables back up to the caller.
```

Comment:
I still don't quite understand this. Can you add more to this comment to explain why the parent scope needs to be populated with these variables?

Reply:
Re-read John's comment on your other PR and I understand now. Thanks!

Reply:
This is a workaround for tests (e.g., mbed, libunwind) that include source files from 3rdparty. To compile those files on Windows, we need to use clangw instead of cl. Currently we have maybe_build_clangw(target_file) script that updates all the compilation options for clangw. Since these tests tend to include multiple unit tests from the 3rdparty, the approach is to use a cmake function that iterates over all the tests. However, maybe_build_clangw can only set variables up to the scope of this function, so we need to populate those variables once more.
(See John's comment as well: https://github.com/openenclave/openenclave/pull/2400#discussion_r362641650)